chain/network/src/peer_manager/tests/tier2.rs

```diff
@@ -0,0 +1,211 @@
+use crate::network_protocol::testonly as data;
+use crate::peer_manager::network_state::RECENT_OUTBOUND_CONNECTIONS_MIN_DURATION;
+use crate::peer_manager::network_state::RECONNECT_ATTEMPT_INTERVAL;
+use crate::peer_manager::peer_manager_actor::UPDATE_RECENT_OUTBOUND_CONNECTIONS_INTERVAL;
+use crate::peer_manager::testonly::start as start_pm;
+use crate::peer_manager::testonly::ActorHandler;
+use crate::tcp;
+use crate::testonly::make_rng;
+use crate::time;
+use near_o11y::testonly::init_test_logger;
+use near_primitives::network::PeerId;
+use near_store::db::TestDB;
+use std::sync::Arc;
+
+async fn check_recent_outbound_connections(pm: &ActorHandler, wanted: Vec<PeerId>) {
+    let got: Vec<PeerId> = pm
+        .with_state(move |s| async move {
+            s.get_recent_outbound_connections().iter().map(|c| c.peer_info.id.clone()).collect()
+        })
+        .await;
+
+    assert!(got == wanted, "Expected {:?} but got {:?}", wanted, got);
+}
+
+#[tokio::test]
+async fn test_recent_outbound_connection() {
+    init_test_logger();
+    let mut rng = make_rng(921853233);
+    let rng = &mut rng;
+    let mut clock = time::FakeClock::default();
+    let chain = Arc::new(data::Chain::make(&mut clock, rng, 10));
+
+    let pm0 = start_pm(clock.clock(), TestDB::new(), chain.make_config(rng), chain.clone()).await;
+    let pm1 = start_pm(clock.clock(), TestDB::new(), chain.make_config(rng), chain.clone()).await;
+    let pm2 = start_pm(clock.clock(), TestDB::new(), chain.make_config(rng), chain.clone()).await;
+
+    let id0 = pm0.cfg.node_id();
+    let id1 = pm1.cfg.node_id();
+
+    // connect pm0 to pm1
+    pm0.connect_to(&pm1.peer_info(), tcp::Tier::T2).await;
+    pm2.connect_to(&pm0.peer_info(), tcp::Tier::T2).await;
+
+    clock.advance(RECENT_OUTBOUND_CONNECTIONS_MIN_DURATION);
+    clock.advance(UPDATE_RECENT_OUTBOUND_CONNECTIONS_INTERVAL);
+
+    // check that pm0 stores the outbound connection to pm1
+    check_recent_outbound_connections(&pm0, vec![id1.clone()]).await;
+
+    // check that pm1 does not store anything, as it has no outbound connections
+    check_recent_outbound_connections(&pm1, vec![]).await;
+
+    // check that pm2 stores the outbound connection to pm0
+    check_recent_outbound_connections(&pm2, vec![id0.clone()]).await;
+
+    drop(pm0);
+    drop(pm1);
+    drop(pm2);
```

Comment:
you don't have to put these here any more - since we are using "cargo nextest" now, we will be able to see the real issue in the CI logs in case the test gets flaky.